### DISCUSSION TITLE
How to smoothly upgrade tekton from 0.59 to 0.63

### DISCUSSION THREAD

**Discussion by garyyang85** [tektoncd/operator#1251] [Q&A]
This should be a general question. I use the "oc apply -f " command to install tekton 0.59 on openshift. Now I need a new function in tknton 0.63. How can I smoothly upgrade? 
I can delete the 0.59 first to use the "oc delete -f ", but this will delete all the resources like pipeline / pipelinerun / task. Is there a way to not delete these resources and upgrade to another version? Thanks.

**Comment by vdemeester** [comment]
Hey @garyyang85,

When you say `oc apply -f`, which payload did you install ? `The `release.yaml` from `tektoncd/pipeline` *or* the one from `tektoncd/operator` ?

You should be able to apply the `release.yaml` from `tektoncd/operator` *if* this is the way you used to install tekton (pipeline). If you were using the `release.yaml` from `tektoncd/pipeline`, it might not be super smooth, in any case.

**Reply by garyyang85** [reply]
Thanks for you reply. I use the release.yaml from tektoncd/operator, so just simply "oc apply -f " the new version should be OK.  Will have a try. :)
